server/cmwell-cons/src/main/scala/Upgrade.scala

```diff
@@ -0,0 +1,56 @@
+/**
+  * Copyright 2015 Thomson Reuters
+  *
+  * Licensed under the Apache License, Version 2.0 (the “License”); you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *   http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  * an “AS IS” BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+import semverfi.{SemVersion, Version}
+
+import scala.collection.GenSeq
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object Upgrade {
+
+  class UpgradeFunc (desc: String, func: (GenSeq[String]) => Boolean){
```

Comment:
use case class.

Reply:
func should return `Future[Boolean]`, in line 25 you can wrap `func(hosts)` with an Await.result(..., 1.minute)